schedule/remedy.py

```diff
@@ -0,0 +1,98 @@
+import os
+from pathlib import Path
+from ..utils import *
+from anki.utils import ids2str
+
+
+def RepresentsDatetime(s):
+    try:
+        return datetime.strptime(s, "%Y%m%d")
+    except ValueError:
+        tooltip("Invalid date format")
+        return None
+
+
+def ask_date_range():
+    (s, r) = getText("Enter the start date in the format YYYYMMDD")
+    if not r:
+        return None, None
+    start_date = RepresentsDatetime(s)
+    if not start_date:
+        return None, None
+    (s, r) = getText("Enter the end date in the format YYYYMMDD")
+    if not r:
+        return None, None
+    end_date = RepresentsDatetime(s)
+    if not end_date:
+        return None, None
+    return start_date, end_date
+
+
+def remedy_hard_misuse(did):
+    start_date, end_date = ask_date_range()
+    if not start_date or not end_date:
+        return
+
+    revlog_ids = mw.col.db.list(
+        f"""SELECT id
+        FROM revlog
+        WHERE ease = 2
+        AND id > {start_date.timestamp() * 1000}
+        AND id < {end_date.timestamp() * 1000}
+        """
+    )
+
+    if len(revlog_ids) == 0:
+        tooltip("There are no reviews with a Hard rating in the selected range of dates.")
+        return
+
+    yes = askUser(
+        f"""Between {start_date.strftime("%Y-%m-%d")} and {end_date.strftime("%Y-%m-%d")}, {len(revlog_ids)} reviews had a Hard rating.
+These ratings will be replaced with Again. A copy of the unmodified revlogs will be stored in a CSV file in the addon folder to allow undoing the changes.
+Do you want to proceed?
+    """
+    )
+
+    if not yes:
+        return
+
+    mw.col.db.execute(
+        f"""UPDATE revlog
+        SET ease = 1
+        WHERE id IN {ids2str(revlog_ids)}
+        """
+    )
+
```

Comment:
```suggestion
    mw.set_schema_modified()

```

Please check if this works correctly or needs some modification (like using self.set_schema_modified() instead).

Reply:
we don't need to do the one-way sync. Only modified revlogs need sync.

Reply:
I tried using the latest version (https://github.com/open-spaced-repetition/fsrs4anki-helper/pull/455/commits/c29752cec806f35e49166341851412f4cbd7a5f4) and the modified revlogs weren't synced to the other device.

As far as I know, revlogs once created can't be modified by sync unless a one-way sync is performed.

Reply:
Thanks for the report. I forget to check it in my phone. Seems the modified revlogs have been uploaded, but the other device doesn't update it during sync. So I have to use `set_schema_modified()`.